src/retry/async.rs

```diff
@@ -256,17 +227,17 @@ async fn f() -> Result<(), backoff::Error<&'static str>> {
 
 # #["#,stringify!($rt),"::main]
 # async fn main() {
-let err = backoff::",stringify!($rt),r#"::retry(ExponentialBackoff::default(), f).await.err().unwrap();
+let err = backoff::",stringify!($rt),r#"::retry(ExponentialBackoff::default(), f()).await.err().unwrap();
 assert_eq!(err, "error");
 # }
 ```"#),
                 pub fn retry<I, E, Fn, B>(
                     backoff: B,
                     operation: Fn,
-                ) -> Retry<impl Sleeper, B, NoopNotify, Fn, Fn::Fut>
+                ) -> Retry<impl Sleeper, B, NoopNotify, Fn, I, E>
                 where
                     B: Backoff,
-                    Fn: FutureOperation<I, E>,
+                    Fn: Future<Output = Result<I, Error<E>>>,
```

Comment:
I think you have to have another generic `Fut` here, and then `Fn: FnMut() -> Fut`